v1/topdown/tokens.go

```diff
@@ -1226,7 +1264,61 @@ func getInputSHA(input []byte, h func() hash.Hash) []byte {
 	return hasher.Sum(nil)
 }
 
+type jwtCacheEntry struct {
+	payload        ast.Object
+	header         ast.Object
+	validSignature bool
+}
+
+const tokenCacheName = "io_jwt"
+
+func getTokenFromCache(bctx BuiltinContext, serializedJwt ast.Value, publicKey ast.Value) (bool, ast.Object, ast.Object, bool) {
+	if bctx.InterQueryBuiltinValueCache == nil {
+		return false, nil, nil, false
+	}
+
+	c := bctx.InterQueryBuiltinValueCache.GetCache(tokenCacheName)
+	if c == nil {
+		return false, nil, nil, false
+	}
+
+	key := createTokenCacheKey(serializedJwt, publicKey)
+
+	entry, ok := c.Get(key)
+	if !ok {
+		return false, nil, nil, false
+	}
+
+	if jwtEntry, ok := entry.(jwtCacheEntry); ok {
+		return true, jwtEntry.header, jwtEntry.payload, jwtEntry.validSignature
+	}
+
+	return false, nil, nil, false
+}
+
+func putTokenInCache(bctx BuiltinContext, serializedJwt ast.Value, publicKey ast.Value, header ast.Object, payload ast.Object, validSignature bool) {
+	if bctx.InterQueryBuiltinValueCache == nil {
+		return
+	}
+
+	c := bctx.InterQueryBuiltinValueCache.GetCache(tokenCacheName)
+	if c == nil {
+		return
+	}
+
+	key := createTokenCacheKey(serializedJwt, publicKey)
+
+	c.Insert(key, jwtCacheEntry{header: header, payload: payload, validSignature: validSignature})
+}
+
+func createTokenCacheKey(serializedJwt ast.Value, publicKey ast.Value) ast.Value {
```

Comment:
A comment that explains why these items are used to create the key would be helpful.